### DISCUSSION TITLE
Is it possible to set the filters programmatically to server driven datagrid?

### DISCUSSION THREAD

**Discussion by kumart-netapp** [vmware-clarity/ng-clarity#1519] [Q&A] [answered]
We have a requirement to have a dashboard with alerts types and on click of a alert type, we should be navigate to all alerts page(send this as route data) with the filter state applied prior to first emit event? 

I could not find an example in doc:
https://clarity.design/documentation/datagrid/code/server-driven 

Also does the filters icons marked as filter applied?

I did a small PoC using stackblitz from docs. I could see filtering is working but the filter icons are not reflecting the filter active icon state
Link: https://stackblitz.com/edit/jteg5q?file=src%2Fapp%2Fexample.component.ts

**Comment by kevinbuhmann** [comment, marked as answer]
Yes, use the `[clrFilterValue]` input. https://stackblitz.com/edit/jteg5q-yuvw8a?file=src%2Fapp%2Fexample.component.html

**Reply by kumart-netapp** [reply]
Thanks Kevin. How does this works for custom filter?

**Reply by kevinbuhmann** [reply]
For custom filters, you'll need to use your own API.
